public/app/plugins/panel/geomap/GeomapPanel.tsx

```diff
@@ -467,6 +467,7 @@ export class GeomapPanel extends Component<Props, State> {
     const layers = this.layers.slice(0);
     try {
       const info = await this.initLayer(this.map, newOptions, current.isBasemap);
+      layers[layerIndex]?.handler.dispose?.();
```

Comment:
curious if there will potentially be dangling subscriptions on a layer that doesn't get disposed.  seems dispose only happens when another layer is changed?  maybe "beforefeatureremoved" event ( or some event ) could call dispose.

Reply:
just added it to `componentWillUnmount`!